README.rst
Python Client for Contact Center AI Insights API API
====================================================

|stable| |pypi| |versions|

`Contact Center AI Insights API API`_:  helps users detect and visualize patterns in their contact center data.

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-contact-center-insights.svg
   :target: https://pypi.org/project/google-cloud-contact-center-insights/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-contact-center-insights.svg
   :target: https://pypi.org/project/google-cloud-contact-center-insights/
.. _Contact Center AI Insights API API: https://cloud.google.com/contact-center/insights/docs
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/contactcenterinsights/latest
.. _Product Documentation:  https://cloud.google.com/contact-center/insights/docs

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Contact Center AI Insights API API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Contact Center AI Insights API API.:  https://cloud.google.com/contact-center/insights/docs
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-contact-center-insights


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-contact-center-insights

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Contact Center AI Insights API API
   to see other available methods on the client.
-  Read the `Contact Center AI Insights API API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Contact Center AI Insights API API Product documentation:  https://cloud.google.com/contact-center/insights/docs
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

## [1.4.1](https://github.com/googleapis/python-contact-center-insights/compare/v1.4.0...v1.4.1) (2022-07-29)


### Documentation

* Updating comments ([#215](https://github.com/googleapis/python-contact-center-insights/issues/215)) ([ed3fb16](https://github.com/googleapis/python-contact-center-insights/commit/ed3fb16cd0c08168c719fd0878d6f3e0b03c5924))

## [1.4.0](https://github.com/googleapis/python-contact-center-insights/compare/v1.3.2...v1.4.0) (2022-07-16)


### Features

* add audience parameter ([1a9e2b0](https://github.com/googleapis/python-contact-center-insights/commit/1a9e2b07b7278de2efb72d774157fb40729d4ced))


### Bug Fixes

* **deps:** require google-api-core>=1.32.0,>=2.8.0 ([#207](https://github.com/googleapis/python-contact-center-insights/issues/207)) ([1a9e2b0](https://github.com/googleapis/python-contact-center-insights/commit/1a9e2b07b7278de2efb72d774157fb40729d4ced))
* require python 3.7+ ([#209](https://github.com/googleapis/python-contact-center-insights/issues/209)) ([864afd4](https://github.com/googleapis/python-contact-center-insights/commit/864afd4131e03ff032fdac58077180956824d0bb))

## [1.3.2](https://github.com/googleapis/python-contact-center-insights/compare/v1.3.1...v1.3.2) (2022-06-07)


### Bug Fixes

* **deps:** require protobuf <4.0.0dev ([#199](https://github.com/googleapis/python-contact-center-insights/issues/199)) ([72b1db8](https://github.com/googleapis/python-contact-center-insights/commit/72b1db85bc851e28a52337bba6ca15f3d5c30f59))


### Documentation

* fix changelog header to consistent size ([#200](https://github.com/googleapis/python-contact-center-insights/issues/200)) ([097bb5c](https://github.com/googleapis/python-contact-center-insights/commit/097bb5c76ba0073fbbd0c8d9ddf5e3e5689a430e))
* Updating comments ([#203](https://github.com/googleapis/python-contact-center-insights/issues/203)) ([9a4888e](https://github.com/googleapis/python-contact-center-insights/commit/9a4888efb120e2d061207217a2f9b5df995eea94))

## [1.3.1](https://github.com/googleapis/python-contact-center-insights/compare/v1.3.0...v1.3.1) (2022-03-07)


### Bug Fixes

* **deps:** require google-api-core>=1.31.5, >=2.3.2 ([#136](https://github.com/googleapis/python-contact-center-insights/issues/136)) ([dc326bc](https://github.com/googleapis/python-contact-center-insights/commit/dc326bc2d8c9f5b49041a05eb2aa4dcb77e332d2))

## [1.3.0](https://github.com/googleapis/python-contact-center-insights/compare/v1.2.1...v1.3.0) (2022-02-26)


### Features

* add api key support ([#126](https://github.com/googleapis/python-contact-center-insights/issues/126)) ([976ac64](https://github.com/googleapis/python-contact-center-insights/commit/976ac649548ed78ccb4f4ad562104a66f4bf77fa))


### Bug Fixes

* resolve DuplicateCredentialArgs error when using credentials_file ([02ad5e1](https://github.com/googleapis/python-contact-center-insights/commit/02ad5e18a4e95e6c4d57e2dbc823e9c9505f19fd))

## [1.2.1](https://github.com/googleapis/python-contact-center-insights/compare/v1.2.0...v1.2.1) (2022-01-20)


### Documentation

* Clarify comments of ConversationView enum ([#123](https://github.com/googleapis/python-contact-center-insights/issues/123)) ([96196d9](https://github.com/googleapis/python-contact-center-insights/commit/96196d9ad14bddbb4ddb70e19ae5add9a20d9cbd))

## [1.2.0](https://github.com/googleapis/python-contact-center-insights/compare/v1.1.0...v1.2.0) (2022-01-14)


### Features

* Add WriteDisposition to BigQuery Export API ([#107](https://github.com/googleapis/python-contact-center-insights/issues/107)) ([bb36139](https://github.com/googleapis/python-contact-center-insights/commit/bb361392935268ba9c45bf89e71876dc0132fc5a))
* new API for the View resource ([#114](https://github.com/googleapis/python-contact-center-insights/issues/114)) ([e55463c](https://github.com/googleapis/python-contact-center-insights/commit/e55463cb32d988273bf328fbc16394e64dd946d5))

## [1.1.0](https://www.github.com/googleapis/python-contact-center-insights/compare/v1.0.0...v1.1.0) (2021-11-05)


### Features

* Add ability to update phrase matchers ([#93](https://www.github.com/googleapis/python-contact-center-insights/issues/93)) ([0b5b07a](https://www.github.com/googleapis/python-contact-center-insights/commit/0b5b07a2b1747c567cce0baea481db07403dc465))
* Add display name to issue model stats ([0b5b07a](https://www.github.com/googleapis/python-contact-center-insights/commit/0b5b07a2b1747c567cce0baea481db07403dc465))
* Add issue model stats to time series ([0b5b07a](https://www.github.com/googleapis/python-contact-center-insights/commit/0b5b07a2b1747c567cce0baea481db07403dc465))

## [1.0.0](https://www.github.com/googleapis/python-contact-center-insights/compare/v0.6.1...v1.0.0) (2021-11-03)


### Features

* bump release level to production/stable ([#89](https://www.github.com/googleapis/python-contact-center-insights/issues/89)) ([d93e829](https://www.github.com/googleapis/python-contact-center-insights/commit/d93e829c10a39c67970edb4f89e55bb39f5ae5a0))

## [0.6.1](https://www.github.com/googleapis/python-contact-center-insights/compare/v0.6.0...v0.6.1) (2021-11-02)


### Bug Fixes

* **deps:** drop packaging dependency ([daf9af8](https://www.github.com/googleapis/python-contact-center-insights/commit/daf9af84fc1b448f86d340c56bc37460b0254f82))
* **deps:** require google-api-core >= 1.28.0 ([daf9af8](https://www.github.com/googleapis/python-contact-center-insights/commit/daf9af84fc1b448f86d340c56bc37460b0254f82))


### Documentation

* list oneofs in docstring ([daf9af8](https://www.github.com/googleapis/python-contact-center-insights/commit/daf9af84fc1b448f86d340c56bc37460b0254f82))

## [0.6.0](https://www.github.com/googleapis/python-contact-center-insights/compare/v0.5.0...v0.6.0) (2021-10-11)


### Features


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2022 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

ALL_PYTHON = [
    "3.6",
    "3.7",
    "3.8",
    "3.9",
    "3.10",
]

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")

BLACK_VERSION = "black==19.10b0"
BLACK_PATHS = ["docs", "google", "tests", "samples", "noxfile.py", "setup.py"]
DEFAULT_PYTHON_VERSION = "3.9"

nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
    "blacken",
    "lint",
    "lint_setup_py",
]

@nox.session(python=ALL_PYTHON)
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/contact_center_insights_v1/',
        '--cov=tests/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=ALL_PYTHON)
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python=DEFAULT_PYTHON_VERSION)
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx==4.0.1", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint(session):
    """Run linters.

    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION)
    session.run(
        "black",
        "--check",
        *BLACK_PATHS,
    )
    session.run("flake8", "google", "tests", "samples")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def blacken(session):
    """Run black. Format code to uniform standard."""
    session.install(BLACK_VERSION)
    session.run(
        "black",
        *BLACK_PATHS,
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
